fennel/client/client.py

```diff
@@ -403,9 +437,58 @@ def extract_features(
                 )
         return df
 
+    def extract_historical(
+        self,
+        inputs: List[Union[Feature, str]],
+        outputs: List[Union[Feature, Featureset, str]],
+        timestamp_column: str,
+        format: str = "pandas",
+        input_dataframe: Optional[pd.DataFrame] = None,
+        input_s3: Optional[S3Connector] = None,
+        output_s3: Optional[S3Connector] = None,
+        feature_to_column_map: Optional[Dict[Feature, str]] = None,
+    ) -> Dict[str, Any]:
+        """
+        Extract point in time correct features from a dataframe, where the
+        timestamps are provided by the timestamps parameter.
+
+        Parameters:
+        inputs (List[Feature]): List of features or featuresets to use as input.
+        outputs (List[Union[Feature, Featureset]]): List of features or featuresets to compute.
+        timestamp_column (str): The name of the column containing the timestamps.
+        format (str): The format of the input data. Can be either "pandas",
+            "csv", "json" or "parquet". Default is "pandas".
+        input_dataframe (Optional[pd.DataFrame]): Dataframe containing the input features. Only relevant when format is "pandas".
+        output_s3 (Optional[S3Connector]): Contains the S3 info -- bucket, prefix, and optional access key id
+            and secret key -- used for storing the output of the extract historical request
+
+        The following parameters are only relevant when format is "csv", "json" or "parquet".
+
+        input_s3 (Optional[sources.S3Connector]): The info for the input S3 data, containing bucket, prefix, and optional access key id
+            and secret key
+        feature_to_column_map (Optional[Dict[Feature, str]]): A dictionary that maps columns in the S3 data to the required features.
+
+
+        Returns:
+        Dict[str, Any]: A dictionary containing the request_id, the output s3 bucket and prefix, the completion rate and the failure rate.
+                        A completion rate of 1.0 indicates that all processing has been completed.
+                        A failure rate of 0.0 indicates that all processing has been completed successfully.
+                        The status of the request.
+        """
+        return self.extract_historical_features(
```

Comment:
I would suggest to do the reverse, let the new API contain the actual logic and the older API call the newer API, so that later we can simply delete the older API.

Reply:
done.